src/lib/orion_websocket/ws.cpp

```diff
@@ -105,9 +110,26 @@ static int wscllbck(libwebsocket_context *ctx,
 
     case LWS_CALLBACK_RECEIVE:
     {
-      ConnectionInfo *ci = connection_manager_get(dat->cid, message);
-      dat->message = strdup((restService(ci, orionServices)).c_str());
-      libwebsocket_callback_on_writable(ctx, ws);
+      const size_t remaining = lws_remaining_packet_payload(ws);
+      const size_t msg_size = strlen(message);
+
+      if (!dat->request)
+        dat->request = (char *) malloc (msg_size + remaining);
+
+      memcpy(&dat->request[dat->index], message, msg_size);
```

Comment:
Normally you would include the terminating ZERO of a string.
If not, strange things can happen ...

[ I say this as msg_size comes from strlen(message) ]